weave/wandb_interface/wandb_stream_table.py

```diff
@@ -136,57 +63,78 @@ def __init__(
         # technically, we could infer the entity from the API key, but
         # that tends to confuse users.
         if entity_name is None or entity_name == "":
-            raise ValueError(f"Must specify entity_name")
+            raise ValueError("Must specify entity_name")
         elif project_name is None or project_name == "":
-            raise ValueError(f"Must specify project_name")
+            raise ValueError("Must specify project_name")
         elif table_name is None or table_name == "":
-            raise ValueError(f"Must specify table_name")
+            raise ValueError("Must specify table_name")
 
         self._lite_run = InMemoryLazyLiteRun(
             entity_name,
             project_name,
             table_name,
+            config=config,
             group="weave_stream_tables",
-            _hide_in_wb=True,
-            _use_async_file_stream=not _disable_async_file_stream,
+            _hide_in_wb=hidden,
         )
+        if not wandb_public_api().supports_streamtable:
+            raise errors.Error(
+                "Streamtable isn't supported in this version of wandb.  Contact your adminstrator to upgrade."
+            )
         self._table_name = self._lite_run._run_name
         self._project_name = self._lite_run._project_name
         self._entity_name = self._lite_run._entity_name
         self._ensure_remote_initialized()
         atexit.register(self._at_exit)
 
-    def _ensure_remote_initialized(self) -> StreamTableType:
-        self._lite_run.ensure_run()
-        print_url = False
-        if not hasattr(self, "_weave_stream_table"):
-            print_url = True
-            self._weave_stream_table = StreamTableType(
-                table_name=self._table_name,
-                project_name=self._project_name,
-                entity_name=self._entity_name,
-            )
-            self._weave_stream_table_ref = storage._direct_publish(
-                self._weave_stream_table,
-                name=self._table_name,
-                wb_project_name=self._project_name,
-                wb_entity_name=self._entity_name,
-                _lite_run=self._lite_run,
-            )
-        if not hasattr(self, "_artifact"):
-            uri = runfiles_wandb.WeaveWBRunFilesURI.from_run_identifiers(
-                self._entity_name,
-                self._project_name,
+    def _ensure_remote_initialized(self) -> None:
+        with self._lock:
+            self._lite_run.ensure_run()
+            print_url = False
+            self._artifact = self._stream_table_artifact()
+            if print_url:
+                base_url = wandb_public_api().settings["base_url"]
+                if base_url.endswith("api.wandb.ai"):
+                    base_url = base_url.replace("api", "weave.")
```

Comment:
:puke: god these URLs are ugly.

Looks like we never print the url in this mode - let's remove?